### PR TITLE
Improve punch card

### PR DESCRIPTION
**Pull request by lvdoorn**
The punch card graph is now full-width and shows informative tooltips.

@agudek You may wish to check if I merged main.js correctly.

### CODE REVIEW

**Comment by borek2** [comment]
Nice job Leendert! I had a suggestion in my PR #236 for the punch cards. Is that hard to implement?

**Comment by lvdoorn** [comment]
@borek2 Thanks for your suggestion. It should be quite trivial to implement.

File: src/main.js [hunk 1]
```diff
@@ -4,257 +4,260 @@
...
-                axis.tickSize(720-50-50);
+        function setStandardAxisValues(module, axisname, axis) {
+            var svgSize = octopeerHelper.getSafeModuleValue(module, "customSVGSize");
+            var margin = octopeerHelper.getSafeModuleValue(module, "margin");
+
+            axis.scale().range([svgSize.h - margin.left - margin.top, 0]).nice();
+            if (axisname === "x") {
+
+                axis.orient("bottom");
+                axis.scale().range([svgSize.w - margin.right - margin.bottom, 0]);
```
Comment (agudek) [inline src/main.js]: this chould be svgSize.w - margin.right - margin.left

File: src/main.js [hunk 2]
```diff
@@ -4,257 +4,260 @@
...
+                if (octopeerHelper.getSafeModuleValue(module, axisname + "AxisTicks")) {
+                    axis.tickSize(-svgSize.h + margin.left + margin.top);
```
Comment (agudek) [inline src/main.js]: this should be -svgSize.h + margin.bottom+ margin.top

File: src/main.js [hunk 3]
```diff
@@ -4,257 +4,260 @@
...
+                }
+            } else if (axisname === "y") {
+                axis.orient("left");
+                if (octopeerHelper.getSafeModuleValue(module, axisname + "AxisTicks")) {
+                    axis.tickSize(-svgSize.w + margin.right + margin.bottom);
```
Comment (agudek) [inline src/main.js]: this should be -svgSize.w + margin.right + margin.left

File: src/main.js [hunk 4]
```diff
@@ -4,257 +4,260 @@
...
+                }
+            } else {
+                axis.orient("right");
+                if (octopeerHelper.getSafeModuleValue(module, axisname + "AxisTicks")) {
+                    axis.tickSize(svgSize.w - margin.right - margin.bottom);
```
Comment (agudek) [inline src/main.js]: this should be svgSize.w - margin.right - margin.left

File: src/main.js [hunk 5]
```diff
@@ -4,257 +4,260 @@
...
-            axis.orient("right");
-            if(octopeerHelper.getSafeModuleValue(module,axisname+"AxisTicks")) {
-                axis.tickSize(720-50-50);
+            var svgSize = octopeerHelper.getSafeModuleValue(module, "customSVGSize");
+            var margin = octopeerHelper.getSafeModuleValue(module, "margin");
+
+            if (axisname === "x") {
+                axis.orient("bottom");
+                if (octopeerHelper.getSafeModuleValue(module, axisname + "AxisTicks")) {
+                    axis.tickSize(-svgSize.h + margin.left + margin.bottom);
```
Comment (agudek) [inline src/main.js]: this should be -svgSize.h + margin.top+ margin.bottom

File: src/main.js [hunk 6]
```diff
@@ -4,257 +4,260 @@
...
+                }
+            } else if (axisname === "y") {
+                axis.orient("left");
+                if (octopeerHelper.getSafeModuleValue(module, axisname + "AxisTicks")) {
+                    axis.tickSize(-svgSize.w + margin.right + margin.top);
```
Comment (agudek) [inline src/main.js]: this should be -svgSize.w + margin.right + margin.left

File: src/main.js [hunk 7]
```diff
@@ -4,257 +4,260 @@
...
+                }
+            } else {
+                axis.orient("right");
+                if (octopeerHelper.getSafeModuleValue(module, axisname + "AxisTicks")) {
+                    axis.tickSize(svgSize.w - margin.left - margin.bottom);
```
Comment (agudek) [inline src/main.js]: this should be svgSize.w - margin.left - margin.right

**Pull request merged**